### PR TITLE
adding perl modules and screen

### PR DESCRIPTION
**Pull request by smartmarmot**

### CODE REVIEW

**Comment by smartmarmot** [comment]
This is a set of perl-module and the screen utility.

File: screen/screen-4.2.1.patch [hunk 1]
```diff
@@ -0,0 +1,54 @@
+--- screen-4.2.1/screen.c	2014-04-26 12:22:09.000000000 -0400
++++ screen-4.2.1/screen.c	2016-05-10 11:33:42.443307000 -0400
+@@ -1109,8 +1109,10 @@
+ 	      n = (eff_uid == 0 && (real_uid || (st.st_mode & 0775) != 0775)) ? 0755 :
+ 	          (eff_gid == (int)st.st_gid && eff_gid != real_gid) ? 0775 :
+ 		  0777;
+-	      if (((int)st.st_mode & 0777) != n)
++#if !defined(__CYGWIN__) || !defined(__MSYS__)
```
Comment (Alexpux) [inline screen/screen-4.2.1.patch]: __MSYS__ is extra here as it __CYGWIN__ defined for MSYS

File: screen/screen-4.2.1.patch [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ 		Panic(0, "Directory '%s' must have mode %03o.", SockDir, n);
++#endif
+ 	    }
+ 	  sprintf(SockPath, "%s/S-%s", SockDir, LoginName);
+ 	  if (access(SockPath, F_OK))
+@@ -1140,8 +1142,10 @@
+       if ((int)st.st_uid != real_uid)
+ 	Panic(0, "You are not the owner of %s.", SockPath);
+     }
++#if !defined(__CYGWIN__) || !defined(__MSYS__)
```
Comment (Alexpux) [inline screen/screen-4.2.1.patch]: __MSYS__ is extra here as it __CYGWIN__ defined for MSYS

File: screen/screen-4.2.1.patch [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
++# if defined(GETUTENT) && (!defined(SVR4) || defined(__hpux)) && ! defined(__CYGWIN__) && ! defined(__MSYS__)
+ #  if defined(hpux) /* cruel hpux release 8.0 */
+ #   define pututline _pututline
+ #  endif /* hpux */
+@@ -586,7 +586,7 @@
+     }
+ #endif
+   setutent();
+-#ifndef __CYGWIN__
++#if !defined(__CYGWIN__) || ! defined (__MSYS__)
```
Comment (Alexpux) [inline screen/screen-4.2.1.patch]: __MSYS__ is extra here as it __CYGWIN__ defined for MSYS

File: screen/screen-4.2.1.patch [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+   return pututline(u) != 0;
+ #else
+   return 1;
+@@ -598,7 +598,7 @@
+ struct utmp *u;
+ {
+   u->ut_type = DEAD_PROCESS;
+-#if (!defined(linux) || defined(EMPTY)) && !defined(__CYGWIN__)
++#if (!defined(linux) || defined(EMPTY)) && !defined(__CYGWIN__) && ! defined (__MSYS__)
```
Comment (Alexpux) [inline screen/screen-4.2.1.patch]: __MSYS__ is extra here as it __CYGWIN__ defined for MSYS

**Comment by smartmarmot** [comment]
ah so msys contains defined also the "CYGWIN" defines?

**Pull request closed without merging**